model-optimizer/mo/front/tf/extractors/subgraph_utils.py

```diff
@@ -0,0 +1,106 @@
+    for pb_node in subgraph_proto['node_def']:
+        # create an NX node
+        id = body_graph.unique_id(pb_node.name)
+        map_original_name[pb_node.name] = id
+        body_graph.add_node(id, pb=pb_node, kind='op')
+        if hasattr(body_graph, 'op_names_statistic') and hasattr(pb_node, 'op'):
+            body_graph.op_names_statistic[pb_node.op] += 1
+
+        # add incoming edges based on data_nodes_map
+        for dst_port, inp in enumerate(pb_node.input):
+            orig_src_id = inp.split(":")[0]
+
+            # TODO: avoid this temporal workaround for TF 2.4 or higher RNN layers:
+            #  skip control flow dependency
+            if orig_src_id[0] == '^':
+                continue
+
+            src_id = map_original_name[orig_src_id]
+            src_port = 0 if len(inp.split(":")) == 1 else int(inp.split(":")[-1])
+            assert (body_graph.has_node(src_id))
+
+            body_graph.add_edges_from([create_tf_edge(src_id + ":" + str(src_port), id, dst_port)])
+
+    # create Result nodes in the loop body graph
+    for output in subgraph_proto['output_arg']:
+        output_name = subgraph_proto['ret'][output.name]
+        orig_src_id = output_name.split(":")[0]
+        src_id = map_original_name[orig_src_id]
+        src_port = 0 if len(output_name.split(":")) == 1 \
+            else int(output_name.split(":")[-1])
+        assert body_graph.has_node(src_id), 'The body graph does not contain output with name "{}"'.format(
+            src_id)
+        body_results.append(Node(body_graph, add_opoutput(body_graph, src_id, src_port, False)))
+
+
+def get_graph_proto(external_graph: Graph, graph_id: str, node_with_graph: Node):
+    graph_name = node_with_graph.pb.attr[graph_id].func.name
+    node_name = node_with_graph.soft_get('name', node_with_graph.id)
+
+    assert 'library' in external_graph.graph, 'The graph does not contain a library that is required ' \
+                                              'by node with name "{}".'.format(node_name)
+
+    library_graph = external_graph.graph['library']
+
+    assert graph_name in library_graph, 'The library does not contain a function with name "{}" ' \
+                                        'that is required by node ' \
+                                        'with name "{}".'.format(graph_name, node_name)
+    return library_graph[graph_name]
+
+
+def create_internal_graph(external_graph: Graph, node_with_internal_graph: Node, internal_graph_name: str):
+    internal_graph = Graph()
+    # fill the body graph
+    for attr_key in external_graph.graph.keys():
+        if attr_key != 'library':
+            internal_graph.graph[attr_key] = copy.deepcopy(external_graph.graph[attr_key])
+        else:
+            # it is sufficient to have a link to the library
+            internal_graph.graph['library'] = external_graph.graph['library']
+    node_with_internal_graph[internal_graph_name] = internal_graph
```

Comment:
I think we can return `internal_graph` from this function and assign it to `node_with_internal_graph[internal_graph_name]` outside.

Reply:
done